core/src/main/java/org/elasticsearch/index/mapper/RangeFieldMapper.java

```diff
@@ -427,10 +432,73 @@ protected void doXContentBody(XContentBuilder builder, boolean includeDefaults,
 
     /** Enum defining the type of range */
     public enum RangeType {
+        IP("ip_range") {
+            @Override
+            public Field getRangeField(String name, Range r) {
+                return new InetAddressRange(name, (InetAddress)r.from, (InetAddress)r.to);
+            }
+            @Override
+            public InetAddress parseFrom(RangeFieldType fieldType, XContentParser parser, boolean coerce, boolean included)
+                throws IOException {
```

Comment:
can you indent so that the throws is not indented at the same level as the content of the block? (similar issues on the below lines)